pimd/pim_cmd_common.c

```diff
@@ -0,0 +1,1356 @@
+				pim_if_lan_delay_enabled(ifp) ? "yes" : "no");
+			vty_out(vty, "Effective Propagation Delay  : %d msec\n",
+				pim_if_effective_propagation_delay_msec(ifp));
+			vty_out(vty, "Effective Override Interval  : %d msec\n",
+				pim_if_effective_override_interval_msec(ifp));
+			vty_out(vty, "Join Prune Override Interval : %d msec\n",
+				pim_if_jp_override_interval_msec(ifp));
+			vty_out(vty, "\n");
+			vty_out(vty, "\n");
+
+			vty_out(vty, "LAN Prune Delay\n");
+			vty_out(vty, "---------------\n");
+			vty_out(vty, "Propagation Delay           : %d msec\n",
+				pim_ifp->pim_propagation_delay_msec);
+			vty_out(vty, "Propagation Delay (Highest) : %d msec\n",
+				pim_ifp->pim_neighbors_highest_propagation_delay_msec);
+			vty_out(vty, "Override Interval           : %d msec\n",
+				pim_ifp->pim_override_interval_msec);
+			vty_out(vty, "Override Interval (Highest) : %d msec\n",
+				pim_ifp->pim_neighbors_highest_override_interval_msec);
+			vty_out(vty, "\n");
+			vty_out(vty, "\n");
+
+			vty_out(vty, "BSM Status\n");
+			vty_out(vty, "----------\n");
+			vty_out(vty, "Bsm Enabled          : %s\n",
+				pim_ifp->bsm_enable ? "yes" : "no");
+			vty_out(vty, "Unicast Bsm Enabled  : %s\n",
+				pim_ifp->ucast_bsm_accept ? "yes" : "no");
+			vty_out(vty, "\n");
+			vty_out(vty, "\n");
+		}
+	}
+
+	if (!found_ifname)
+		vty_out(vty, "%% No such interface\n");
+}
+
+void ip_pim_ssm_show_group_range(struct pim_instance *pim, struct vty *vty,
+				 bool uj)
+{
+	struct pim_ssm *ssm = pim->ssm_info;
+	const char *range_str =
+		ssm->plist_name ? ssm->plist_name : PIM_SSM_STANDARD_RANGE;
+
+	if (uj) {
+		json_object *json;
+
+		json = json_object_new_object();
+		json_object_string_add(json, "ssmGroups", range_str);
+		vty_json(vty, json);
+	} else
+		vty_out(vty, "SSM group range : %s\n", range_str);
+}
+
+struct pnc_cache_walk_data {
+	struct vty *vty;
+	struct pim_instance *pim;
+};
+
+static int pim_print_pnc_cache_walkcb(struct hash_bucket *bucket, void *arg)
+{
+	struct pim_nexthop_cache *pnc = bucket->data;
+	struct pnc_cache_walk_data *cwd = arg;
+	struct vty *vty = cwd->vty;
+	struct pim_instance *pim = cwd->pim;
+	struct nexthop *nh_node = NULL;
+	ifindex_t first_ifindex;
+	struct interface *ifp = NULL;
+	char buf[PREFIX_STRLEN];
+
+	for (nh_node = pnc->nexthop; nh_node; nh_node = nh_node->next) {
+		first_ifindex = nh_node->ifindex;
+		ifp = if_lookup_by_index(first_ifindex, pim->vrf->vrf_id);
+
+		vty_out(vty, "%-15s ",
+			inet_ntop(AF_INET, &pnc->rpf.rpf_addr.u.prefix4, buf,
+				  sizeof(buf)));
+		vty_out(vty, "%-16s ", ifp ? ifp->name : "NULL");
+		vty_out(vty, "%pI4 ", &nh_node->gate.ipv4);
+		vty_out(vty, "\n");
+	}
+	return CMD_SUCCESS;
+}
+
+void pim_show_nexthop(struct pim_instance *pim, struct vty *vty)
+{
+	struct pnc_cache_walk_data cwd;
+
+	cwd.vty = vty;
+	cwd.pim = pim;
+	vty_out(vty, "Number of registered addresses: %lu\n",
+		pim->rpf_hash->count);
+	vty_out(vty, "Address         Interface        Nexthop\n");
```

Comment:
For IPV6 address we might need more space.

Please check every show command where we are displaying address